### PR TITLE
ATOM-TYPE-20260120-AWI-DSpy-Toolkit

### PR DESCRIPTION
**Pull request by toolate28**
This pull request introduces a comprehensive prompt generation toolkit for AWI, inspired by DSPy, and adds robust testing, documentation, and validation improvements. The main focus is on integrating a modular prompt generation pipeline, enhancing security and correctness in scripts, and providing thorough unit tests for edge cases and optimizations.

**Prompt Generation Toolkit Integration:**

- Added a new section to `interface/awi-spec.md` documenting the AWI prompt toolkit, including its architecture, components (`ChainOfThought`, `Predict`, `AwiPromptGen`), optimization techniques (COPRO, SIMBA), and usage examples. Reference to the implementation in `experiments/awi_prompt_gen.py` is also provided.
- Updated `requirements-ml.txt` to mention the optional `dspy-ai` dependency for prompt optimization, clarifying its use for the new AWI prompt toolkit.

**Testing and Quality Assurance:**

- Added a comprehensive unit test suite in `experiments/test_awi_prompt_gen.py` covering:
  - Integration between scaffolder and refiner
  - Metadata propagation
  - Handling of empty/malformed intents and histories
  - YAML injection prevention
  - Permission level inference
  - Coherence example validation for COPRO optimization

**Security and Validation Enhancements:**

- Improved input validation in the `awi_request` function within `ops/scripts/spiralsafe` by ensuring `level` and `ttl` are non-negative integers, preventing malformed requests and potential script errors.

**Maintenance and Cleanup:**

- Removed outdated quantum cognition session history data from `media/output/quantum_cognition/session_history.json`, likely as part of a cleanup or data refresh.This pull request introduces a comprehensive suite of improvements focused on prompt generation for AI-human collaboration, validation enhancements, and documentation updates. The most significant changes are the addition of a unit test module for the AWI prompt generation system, new documentation for the prompt toolkit architecture and usage, stricter input validation in the `spiralsafe` script, and optional support for DSPy-style prompt optimization.

**Prompt Generation Toolkit and Testing**

* Added a full unit test suite for the `AwiPromptGen` module in `experiments/test_awi_prompt_gen.py`, covering integration, metadata propagation, edge case handling, YAML injection prevention, permission inference, and coherence example validation.
* Documented the AWI prompt toolkit architecture and usage in `interface/awi-spec.md`, including the DSPy-inspired pipeline, optimization techniques (COPRO, SIMBA), and a reference implementation.
* Updated `requirements-ml.txt` to optionally support DSPy-style prompt optimization by adding a commented-out dependency for `dspy-ai`.

**Validation Improvements**

* Enhanced input validation in the `awi_request` function of `ops/scripts/spiralsafe` to ensure `level` and `ttl` are non-negative integers without leading zeros (except for 0).

**Data Cleanup**

* Removed obsolete quantum cognition session history data from `media/output/quantum_cognition/session_history.json` to keep the repository clean and relevant.name: Pull request template
about: Use this template for PRs that add CI, bump templates, or agent-facing files
title: ''
labels: ''
assignees: ''

---

## Summary

Describe the change in 1-2 sentences.

## ATOM Tag

**ATOM:** `ATOM-TYPE-YYYYMMDD-NNN-description`

_(Generate with: `./scripts/atom-track.sh TYPE "description" "file"`)_

## Why

Why this change is needed and what it enables.

## What changed

- Files added
- Scripts
- CI workflows

## Verification / Testing

- [ ] `scripts/validate-bump.sh` passes locally (if bump.md changed)
- [ ] `scripts/validate-branch-name.sh` tested on example branches (if applicable)
- [ ] `bash scripts/verify-environment.sh` prints `ENV OK`
- [ ] `scripts/test-scripts.sh` passes (if scripts changed)
- [ ] All shell scripts pass shellcheck
- [ ] ATOM tag created and logged

## Claude Interaction

You can interact with Claude in this PR by:

- **@mentioning Claude** in comments for questions or reviews
- **Adding labels**: `claude:review`, `claude:help`, `claude:analyze`
- **Requesting reviews**: Claude will provide automated feedback
- **Ask questions**: Claude can explain code, suggest improvements, or identify issues

### Example commands:

- `@claude please review this PR for ATOM compliance`
- `@claude explain the changes in scripts/atom-track.sh`
- `@claude check for security issues`
- `@claude suggest improvements`

### CODE REVIEW

File: experiments/awi_prompt_gen.py [hunk 1]
```diff
@@ -0,0 +1,715 @@
...
+    │ Optimized       │ <-- │ Predict         │ <-- │ History Context │
+    │ Prompt          │     │ Refiner         │     │ (past sessions) │
+    └─────────────────┘     └─────────────────┘     └─────────────────┘
+
+Integration Points:
+- wave-toolkit: Coherence detection for prompt quality
+- AWI protocol: Permission scaffolding integration
+- ATOM system: Session tracking and verification
+
+ATOM-FEATURE-20260117-001-awi-prompt-gen
```
[the remark this answers is not in the document]
Comment (toolate28) [reply experiments/awi_prompt_gen.py:44]: @copilot

File: experiments/test_awi_prompt_gen.py [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    # Check positive examples have high coherence
+    for ex in examples.get("positive", []):
+        assert ex.coherence_score >= COHERENCE_HIGH_THRESHOLD, \
+            f"Positive example should have high coherence: expected >= {COHERENCE_HIGH_THRESHOLD}, got {ex.coherence_score}"
+        assert ex.is_positive is True, "Positive example should have is_positive=True"
+
+    # Check negative examples have low coherence
+    for ex in examples.get("negative", []):
+        assert ex.coherence_score < COHERENCE_HIGH_THRESHOLD, \
+            f"Negative example should have low coherence: expected < {COHERENCE_HIGH_THRESHOLD}, got {ex.coherence_score}"
```
[the remark this answers is not in the document]
Comment (toolate28) [reply experiments/test_awi_prompt_gen.py:161]: @claude

File: experiments/test_awi_prompt_gen.py [hunk 3]
```diff
@@ -0,0 +1,163 @@
+#!/usr/bin/env python3
```
[the remark this answers is not in the document]
Comment (toolate28) [reply experiments/test_awi_prompt_gen.py:1]: @copilot
